packages/e2e-tests/specs/editor/various/block-directory-add.test.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createNewPost,
+	searchForBlock,
+	setUpResponseMocking,
+	getEditedPostContent,
+	getAllBlocks,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { enableExperimentalFeatures } from '../../../experimental-features';
+
+// Urls to mock
+const SEARCH_URLS = [
+	'/__experimental/block-directory/search',
+	`rest_route=${ encodeURIComponent(
+		'/__experimental/block-directory/search'
+	) }`,
+];
+
+const INSTALL_URLS = [
+	'/__experimental/block-directory/install',
+	`rest_route=${ encodeURIComponent(
+		'/__experimental/block-directory/install'
+	) }`,
+];
+
+// Example Blocks
+const mockBlock1 = {
+	name: 'block-directory-test-block/main-block',
+	title: 'Block Directory Test Block',
+	description: 'This plugin is useful for the block.',
+	id: 'block-directory-test-block',
+	rating: 0,
+	rating_count: 0,
+	active_installs: 0,
+	author_block_rating: 0,
+	author_block_count: 1,
+	author: 'No Author',
+	icon: 'block-default',
+	assets: [
+		'fake_url.com/block.js', // we will mock this
+	],
+	humanized_updated: '5 months ago',
+};
+
+const mockBlock2 = {
+	...mockBlock1,
+	name: 'block-directory-test-block/secondary-block',
+	title: 'Block Directory Test Block - Pt Deux',
+	id: 'block-directory-test-secondary-block',
+};
+
+// Block that will be registered
+const block = `( function() {
+	var registerBlockType = wp.blocks.registerBlockType;
+	var el = wp.element.createElement;
+
+	registerBlockType( '${ mockBlock1.name }', {
+		title: 'Test Block for Block Directory',
+		icon: 'hammer',
+		category: 'common',
+		attributes: {},
+		edit: function( props ) {
+			return el( 'p', null, 'Test Copy' );
+		},
+		save: function() {
+			return null;
+		},
+	} );
+} )();`;
+
+function getResponseObject( obj, contentType ) {
+	return {
+		status: 200,
+		contentType,
+		body: obj,
+	};
+}
+
+function createResponse( mockResponse, contentType = undefined ) {
+	return async ( request ) =>
+		request.respond( getResponseObject( mockResponse, contentType ) );
+}
+
+const matchUrl = ( reqUrl, urls ) => {
+	return urls.some( ( el ) => reqUrl.indexOf( el ) >= 0 );
+};
+
+describe( 'adding blocks from block directory', () => {
+	beforeEach( async () => {
+		await enableExperimentalFeatures( [ '#gutenberg-block-directory' ] );
+		await createNewPost();
+	} );
+
+	it( 'Should show an empty state when no plugin is found.', async () => {
+		// Be super weird so there won't be a matching block installed
+		const impossibleBlockName = '@#$@@Dsdsdfw2#$@';
+
+		// Return an empty list of plugins
+		await setUpResponseMocking( [
+			{
+				match: ( request ) => matchUrl( request.url(), SEARCH_URLS ),
```

Comment:
Yeah, that appears to be a new addition. Looks right, I'll try the change.

Reply:
Since this URL is set to change pretty soon, I would like to suggest we make the change then. It will keep this PR smaller and probably makes more sense in that context. We could alternatively put this PR on ice until the block directory moves away from experiment status but that would deprive the code base from these checks in the interim.

Reply:
okay— i don't want to block getting tests in, so making this change in a following PR sounds good.